Source/JavaScriptCore/wasm/WasmBBQJIT.cpp

```diff
@@ -7292,10 +7292,14 @@ class BBQJIT {
         // Since this can switch instance, we need to keep JSWebAssemblyInstance anchored in the stack.
         m_jit.storePtr(jsCalleeAnchor, Location::fromArgumentLocation(wasmCalleeInfo.thisArgument).asAddress());
 
-        // Safe to use across saveValues/passParameters since neither clobber the scratch GPR.
         m_jit.loadPtr(Address(calleeCode), wasmScratchGPR);
         prepareForExceptions();
-        saveValuesAcrossCallAndPassArguments(arguments, wasmCalleeInfo);
+
+        // This is kind of weird! We can't use wasmScratchGPR here, since it stores the address of callee. But we know
+        // that the calleeCode GPR will never be used after this point, and that it can't be bound to any of our
+        // arguments. So we use it as the scratch GPR for passing the arguments instead.
+        saveValuesAcrossCallAndPassArguments(arguments, wasmCalleeInfo, wasmScratchGPR, wasmScratchFPR);
```

Comment:
💣

Reply:
Yeah I just pushed too quickly...most recent push resolves this and also simplifies and corrects the test. TY for catching!